layouts/_default/_markup/render-image.html

```diff
@@ -1,27 +1,43 @@
 {{- $alt := .PlainText | safeHTML -}}
 {{- $lazyLoading := .Page.Site.Params.enableImageLazyLoading | default true -}}
 {{- $dest := .Destination -}}
+{{- $url := urls.Parse $dest -}}
 
-{{- $isLocal := not (urls.Parse $dest).Scheme -}}
+{{- $isLocal := not $url.Scheme -}}
 {{- $isPage := and (eq .Page.Kind "page") (not .Page.BundleType) -}}
 {{- $startsWithSlash := hasPrefix $dest "/" -}}
 {{- $startsWithRelative := hasPrefix $dest "../" -}}
 
 {{- if and $dest $isLocal -}}
   {{- if $startsWithSlash -}}
-    {{/* Images under static directory */}}
-    {{- $dest = (relURL (strings.TrimPrefix "/" $dest)) -}}
+    {{- with or (.PageInner.Resources.Get $url.Path) (resources.Get $url.Path) -}}
+      {{/* Images under assets directory */}}
+      {{- $query := cond $url.RawQuery (printf "?%s" $url.RawQuery) "" -}}
+      {{- $fragment := cond $url.Fragment (printf "?%s" $url.Fragment) "" -}}
+      {{- $dest = printf "%s%s%s" .RelPermalink $query $fragment -}}
+    {{- else -}}
+      {{/* Images under static directory */}}
+      {{- $dest = (relURL (strings.TrimPrefix "/" $dest)) -}}
+    {{- end -}}
```

Comment:
I saw this in the hugo builtin rener hook 👍 

https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/_default/_markup/render-image.html